Lib/DArray.hpp

```diff
@@ -38,9 +38,6 @@ namespace Lib {
 template<typename C>
 class DArray
 {
-private:
```

Comment:
This looks suspicious - what's going on here? I'm OK with an explicitly-provided `operator=(const DArray &)` but none at all seems suspicious as the compiler-provided one will likely be wrong.

Reply:
I was thinking one is implicitly deleted anyways as we have `DArray& operator=(DArray&& other)...` 
But as we have an implicity copy constructor we still get the copy assignment (before and after the PR).. I fixed this now :)

Reply:
Ahah, I missed the copy constructor later. OK, sounds good - looks like you even managed to eliminate some copies in FMB, nice.